common/src/main/java/com/wynntils/features/ui/ContainerScrollFeature.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright © Wynntils 2023.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.features.ui;
+
+import com.wynntils.core.components.Models;
+import com.wynntils.core.config.Category;
+import com.wynntils.core.config.Config;
+import com.wynntils.core.config.ConfigCategory;
+import com.wynntils.core.config.RegisterConfig;
+import com.wynntils.core.features.Feature;
+import com.wynntils.mc.event.MouseScrollEvent;
+import com.wynntils.utils.mc.McUtils;
+import com.wynntils.utils.wynn.ContainerUtils;
+import net.minecraft.client.gui.screens.Screen;
+import net.minecraft.client.gui.screens.inventory.AbstractContainerScreen;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+import org.lwjgl.glfw.GLFW;
+
+@ConfigCategory(Category.UI)
+public class ContainerScrollFeature extends Feature {
+    private static final int abilityTreePreviousSlot = 57;
+    private static final int abilityTreeNextSlot = 59;
+
+    private static final int bankPreviousSlot = 17; // This works for Misc Bucket, Bank, and Block Bank
+    private static final int bankNextSlot = 8;
+
+    private static final int guildBankPreviousSlot = 9;
+    private static final int guildBankNextSlot = 27;
+
+    private static final int tradeMarketPreviousSlot = 17;
+    private static final int tradeMarketNextSlot = 26;
+
+    @RegisterConfig
+    public final Config<Boolean> invertScroll = new Config<>(false);
+
+    @SubscribeEvent
+    public void onInteract(MouseScrollEvent event) {
+        Screen screen = McUtils.mc().screen;
+
+        if (!(screen instanceof AbstractContainerScreen<?> gui)) return;
+        boolean up = event.isScrollingUp() ^ invertScroll.get();
+
+        int slot;
+
+        if (Models.Container.isAbilityTreeScreen(gui)) {
```

Comment:
I recommend you store the prev/next slot numbers as a pair, then you can separate this into two actions: first get the proper pair depending on the container, and then the a() or b() part of the pair, depending on the scroll direction. Also, please consider moving the slot constants to ContainerModel.